libqtile/backend/base.py

```diff
@@ -41,3 +65,95 @@ def grab_pointer(self) -> None:
     @abstractmethod
     def ungrab_pointer(self) -> None:
         """Release grabbed pointer events"""
+
+    def scan(self) -> None:
+        """Scan for clients if required."""
+
+    def warp_pointer(self, x: int, y: int) -> None:
+        """Warp the pointer to the given coordinates relative."""
+
+    def update_client_list(self, windows_map: Dict[int, WindowType]) -> None:
+        """Update the list of windows being managed"""
+
+    @contextlib.contextmanager
+    def masked(self):
+        """A context manager to suppress window events while operating on many windows."""
+        yield
+
+
+class FloatStates:
```

Comment:
What's the advantage to using that?

Reply:
It can help to convey intent and semantics a bit. If you have a type that accepts or returns an `int`, you need to figure out what that represents, but an annotation of `FloatStates` is more descriptive. Similarly, if you are trying to debug something, printing out the value will be more descriptive as the enum rather than as the int. You also can't do things like accidentally do some arithmetic with the values of an enum like you can with an int.